src/apps/viewer/extensions/cellpack/model.ts

```diff
@@ -237,14 +264,53 @@ async function getIngredientStructure(ingredient: Ingredient, baseUrl: string) {
 
     if (source.pdb === 'None') return
 
-    const model = await getModel(source.pdb || name, baseUrl)
+    const model = await getModel(source.pdb || name,source, baseUrl)
     if (!model) return
 
     if (nbCurve) {
-        return getCurve(name, getCurveTransforms(ingredient), model)
+        return getCurve(name,ingredient, getCurveTransforms(ingredient), model)
     } else {
-        const structure = await getStructure(model, { assembly: source.biomt ? '1' : undefined })
-        return getAssembly(getResultTransforms(results), structure)
+        let bu:string|undefined = source.bu ? source.bu : undefined;
+        if (bu){
+            if (bu=="AU") {bu = undefined;}
+            else {
+                bu=bu.slice(2)
+            }
+        }
+        let structure = await getStructure(model,source, { assembly: bu })
+        //transform with offset and pcp
+        let legacy:boolean = true
+        if (ingredient.offset || ingredient.principalAxis)
+        //center the structure
+        {
+            legacy=false
+            const boundary = computeStructureBoundary(structure)
+            let modelCenter:Vec3 = Vec3.zero()
+            Vec3.scale(modelCenter,boundary.sphere.center,-1.0);//Model.getCenter(structure.models[0])
```

Comment:
there is `Vec3.negate` :)

Also please rename it structureCenter or just center since it is not the model center